spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/NDVSketchGenerator.java

```diff
@@ -0,0 +1,149 @@
+    Map<String, ThetaSketchJavaSerializable> sketchMap = Maps.newHashMap();
+
+    tuple2Iterator.forEachRemaining(tuple -> sketchMap.put(tuple._1, tuple._2));
+    return generateBlobs(table, columnsToBeAnalyzed, sketchMap);
+  }
+
+  private static List<Blob> generateBlobs(
+      Table table, Set<String> columns, Map<String, ThetaSketchJavaSerializable> sketchMap) {
+    return columns.stream()
+        .map(
+            columnName -> {
+              Sketch sketch = sketchMap.get(columnName).getSketch();
+              long ndv = (long) sketch.getEstimate();
+              return new Blob(
+                  StandardBlobTypes.APACHE_DATASKETCHES_THETA_V1,
+                  ImmutableList.of(table.schema().findField(columnName).fieldId()),
+                  table.currentSnapshot().snapshotId(),
+                  table.currentSnapshot().sequenceNumber(),
+                  ByteBuffer.wrap(sketch.toByteArray()),
+                  null,
+                  ImmutableMap.of(APACHE_DATASKETCHES_THETA_V1_NDV_PROPERTY, String.valueOf(ndv)));
+            })
+        .collect(Collectors.toList());
+  }
+
+  static Iterator<Tuple2<String, ThetaSketchJavaSerializable>> computeNDVSketches(
+      SparkSession spark, Table table, long snapshotId, Set<String> toBeAnalyzedColumns) {
+    String tableName = table.name();
+    List<String> columns = ImmutableList.copyOf(toBeAnalyzedColumns);
+    Dataset<Row> data =
+        spark
+            .read()
+            .option(SNAPSHOT_ID, snapshotId)
+            .table(tableName)
+            .select(columns.stream().map(functions::col).toArray(Column[]::new));
+    Schema schema = table.schema();
+    List<Types.NestedField> nestedFields =
+        columns.stream().map(schema::findField).collect(Collectors.toList());
+
+    final JavaPairRDD<String, ByteBuffer> colNameAndSketchPair =
+        data.javaRDD()
+            .flatMap(
+                row -> {
+                  List<Tuple2<String, ByteBuffer>> columnsList =
+                      Lists.newArrayListWithExpectedSize(columns.size());
+                  for (int i = 0; i < row.size(); i++) {
+                    columnsList.add(
+                        new Tuple2<>(
+                            columns.get(i),
+                            Conversions.toByteBuffer(nestedFields.get(i).type(), row.get(i))));
+                  }
+                  return columnsList.iterator();
+                })
+            .mapToPair(t -> t);
+
+    JavaPairRDD<String, ThetaSketchJavaSerializable> sketches =
+        colNameAndSketchPair.aggregateByKey(
+            new ThetaSketchJavaSerializable(),
+            NDVSketchGenerator::update,
+            NDVSketchGenerator::combine);
+
+    return sketches.toLocalIterator();
+  }
+
+  private static ThetaSketchJavaSerializable update(
+      ThetaSketchJavaSerializable sketch, ByteBuffer val) {
+    sketch.update(val);
+    return sketch;
+  }
+
+  private static ThetaSketchJavaSerializable combine(
```

Comment:
I just realize, why cant we move this logic to ThetaSketchJavaSerializable itself?

Then update/combine can just be oneliners, and probably able to be inlined ie :
```
        colNameAndSketchPair.aggregateByKey(
            new ThetaSketchJavaSerializable(),
            ThetaSketchJavaSerializable::update,
            ThetaSketchJavaSerializable::combine,
```